lib/active_model/serializer.rb

```diff
@@ -70,7 +70,7 @@ def self.attribute(attr, options = {})
       ActiveModelSerializers.silence_warnings do
         define_method key do
           object.read_attribute_for_serialization(attr)
-        end unless respond_to?(key, false) || _fragmented.respond_to?(attr)
+        end unless (key != :id && method_defined?(key)) || _fragmented.respond_to?(attr)
```

Comment:
While we're at it, should `_fragmented.respond_to?(attr)` not be `_fragmented.respond_to?(key)`? I haven't delved into the caching yet.